libs/ballot-encoder/src/index.ts

```diff
@@ -79,27 +261,47 @@ export function encodeBallot(
 export function encodeBallotInto(
   election: Election,
   {
-    ballotStyle,
-    precinct,
+    electionHash,
+    ballotStyleId,
+    precinctId,
     votes,
     ballotId,
     isTestMode,
     ballotType,
   }: CompletedBallot,
   bits: BitWriter
 ): BitWriter {
+  const ballotStyle = getBallotStyle({ election, ballotStyleId })
+
+  if (!ballotStyle) {
+    throw new Error(`unknown ballot style id: ${ballotStyleId}`)
+  }
+
   validateVotes({ election, ballotStyle, votes })
 
   const contests = getContests({ ballotStyle, election })
 
   return bits
     .writeUint8(...Prelude)
-    .writeString(ballotStyle.id)
-    .writeString(precinct.id)
-    .writeString(ballotId)
+    .writeString(electionHash.slice(0, ELECTION_HASH_LENGTH), {
```

Comment:
Am i understanding right from the PR description that this new way of encoding the election hash is only for BMD ballots and not HMPB, so this function is only for BMD ballots? Thats pretty hard to understand in the code, but I'm not sure if the intention is for this to be eventually more shared.

Reply:
`encodeBallot` is only for BMD, yeah. Originally that's all there was and the name never changed. Really, there's no _ballot_ data to encode with the HMPB since it's all just metadata, not votes. But these names could maybe be clearer.